Add Nextcloud 30 & 31 support

JonathanTreffler opened the pull request:
### ☑️ Resolves

* Fix #22

This replaces the old removed NC systems (old event system) with their modern equivalents.
I also refactored any code that uses exec() as this command is forbidden and disabled in the environment I want this app to run it and not needed, this can all be done with easy native PHP.

### 🚧 Tasks

- [x] migrate to new event system
- [x] mark as compatible in info.xml
- [x] test simple operations like l10n-override:list and l10n-override:add and if changes actually show up in frontend
- [ ] test more complex operations (disabling, enabling apps, ... ?)

### 🏁 Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] ⛑️ Tests (unit and/or integration) are included or **not required**
- [x] 📘 Documentation has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

JonathanTreffler committed:
composer updates

Signed-off-by: Jonathan Treffler <jonathan.treffler@verdigado.com>

JonathanTreffler committed:
added compatibility with modern nextcloud versions

Signed-off-by: Jonathan Treffler <jonathan.treffler@verdigado.com>

JonathanTreffler commented:
(Added DCO in force-push)

JonathanTreffler committed:
fixed listener path

Signed-off-by: Jonathan Treffler <jonathan.treffler@verdigado.com>

vitormattos commented:
Look the failed checks.

JonathanTreffler committed:
make psalm and php-cs happy

Signed-off-by: Jonathan Treffler <jonathan.treffler@verdigado.com>

vitormattos commented on appinfo/info.xml:
```diff
@@ -35,7 +35,7 @@ To more informations about how this app works or to get examples, look at the ad
 	<repository type="git">https://github.com/librecodecoop/l10n_override</repository>
 	<dependencies>
-		<nextcloud min-version="28" max-version="28"/>
+		<nextcloud min-version="30" max-version="31"/>
```
Another point....

I know that could be a bit complex to make backports but we can set the min and max versionto 32 and after this, send the backport to a branch stable31.

We also can drop Nextcloud 30 because the official support by server will finish into September as we can see here: https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule

JonathanTreffler replied:
Is there any upside to going the backport way ?
I think normally this would be done when code changes to the app are needed between the two versions, so features can be backported to the old branch too.
Since this is only compatibility changes, no feature changes and no code changes between the versions are needed I think this approach would be simpler:
Releasing one version which covers both 30 and 31 and just dropping NC 30 with the a next release after September, when it is no longer supported.

JonathanTreffler replied:
Ah, I just realized I got the PR title wrong. NC 32 is not even out yet, i meant 30 and 31. And AFAIK apps should not mark themselves compatible with a version until that version is at least very close to a release.

JonathanTreffler replied:
Is there any upside to going the backport way ?
I think normally this would be done when code changes to the app are needed between the two versions, so features can be backported to the old branch too.
Since this is only compatibility changes, no feature changes and no code changes between the versions are needed I think this approach would be simpler:
Releasing one version which covers both 30 and 31 and just drop NC 30 with any next release after September, when it is no longer supported.

JonathanTreffler commented on appinfo/info.xml:
```diff
@@ -35,7 +35,7 @@ To more informations about how this app works or to get examples, look at the ad
 	<repository type="git">https://github.com/librecodecoop/l10n_override</repository>
 	<dependencies>
-		<nextcloud min-version="28" max-version="28"/>
+		<nextcloud min-version="30" max-version="31"/>
```
As there are no breaking changes to public interfaces per Semantic Versioning the correct version would be 2.1.0, but I personally would also be OK with 3.0.0, if you prefer that

JonathanTreffler committed:
fix updateAllLanguages

Signed-off-by: Jonathan Treffler <jonathan.treffler@verdigado.com>

JonathanTreffler commented:
@vitormattos Could you please take a look at the updateAllLanguages function ? Before I changed it that function called $this->update() with 3 parameters, even though it only takes two. So I removed the parameter, whose name did not match by name.
But that was still not working as listLanguages does not return a list of languages as the name might imply, but a list of rows from l10n_override_text, so I assume the parameter that should actually be passed to update is the new_language column of that row ? I change that, now it does not throw errors anymore, but you probably know better what it was supposed to do and if this actually does that.
